src/server/pfs/server/http.go

```diff
@@ -84,3 +87,33 @@ func (s *HTTPServer) getFileHandler(w http.ResponseWriter, r *http.Request, ps h
 	}
 	io.Copy(&fw, file)
 }
+
+type loginRequestPayload struct {
+	Token string
+}
+
+func (s *HTTPServer) authLoginHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var data loginRequestPayload
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&data)
+	if err != nil {
+		// Return 500
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(fmt.Sprintf("malformed JSON sent in payload: %v", err)))
```

Comment:
nit: IIUC, go's HTTP library has a convenience function for this: https://golang.org/pkg/net/http/#Error (i.e. `http.Error(w, "malformed JSON", http.StatusInternalServerError)`)